crates/swc_css_modules/tests/fixture.rs

```diff
@@ -0,0 +1,170 @@
+    let mut css_str = String::new();
+    {
+        let wr = BasicCssWriter::new(
+            &mut css_str,
+            None, // Some(&mut src_map_buf),
+            BasicCssWriterConfig::default(),
+        );
+        let mut gen = CodeGenerator::new(wr, CodegenConfig { minify: false });
+        gen.emit(&new_stylesheet).unwrap();
+    }
+    css_str
+}
+
+#[test]
+fn test_css_modules_case() {
+    let simple_compare = |txt: String| -> String {
+        let mut new_str = txt.replace(' ', "");
+        new_str = new_str.trim().replace('\n', "").replace('\r', "");
+        new_str
+    };
+
+    let case_1 = path_resolve("tests/fixture/demo1.css");
+    let res1 = test_css_modules(case_1.as_str());
+    let target_code = r#"
+.x {
+  display: inline-block;
+  width: 20px;
+}
+h2,
+h3 {
+  font-size: 10px;
+  display: block;
+}
+h2 .fixture_demo1_14738004728310336865_a,
+h3 .fixture_demo1_14738004728310336865_a {
+  display: block;
+  box-sizing: border-box;
+}
+h2 .fixture_demo1_14738004728310336865_m .tap h2 ,
+h3 .fixture_demo1_14738004728310336865_m .tap h2  {
+  word-break: break-all;
+  width: 40px;
+}
+.fixture_demo1_14738004728310336865_kol h2 .fixture_demo1_14738004728310336865_m .tap h2 ,
+.fixture_demo1_14738004728310336865_kol h3 .fixture_demo1_14738004728310336865_m .tap h2  {
+  width: 100px;
+}
+.fixture_demo1_14738004728310336865_u h2,
+.fixture_demo1_14738004728310336865_u h3 {
+  display: inline-block;
+  width: 20px;
+}
+h2  .b,
+h3  .b {
+  display: inline-block;
+  width: 20px;
+}
+.fixture_demo1_14738004728310336865_c h2 ,
+.fixture_demo1_14738004728310336865_c h3  {
+  display: inline-block;
+  width: 20px;
+}
+  "#;
+    assert_eq!(
+        simple_compare(target_code.to_string()),
+        simple_compare(res1)
+    );
+
+    let case_2 = path_resolve("tests/fixture/demo2.css");
+    let res2 = test_css_modules(case_2.as_str());
+    let target_code = r#"
+.fixture_demo2_6789844810353091540_a {
+  width: 20px;
+}
+.fixture_demo2_6789844810353091540_x  .a .c,
+.fixture_demo2_6789844810353091540_d  .b {
+  width: 40px;
+}
+.fixture_demo2_6789844810353091540_b .a > .b.c  .fixture_demo2_6789844810353091540_x {
+  width: 30px;
+}
+ .a  .fixture_demo2_6789844810353091540_b,
+ .x  {
+  width: 40px;
+}
+,
+ .fixture_demo2_6789844810353091540_b {
+  width: 40px;
+}
+ .a  .fixture_demo2_6789844810353091540_b,
+ .x  .fixture_demo2_6789844810353091540_m {
+  width: 40px;
+}
+.fixture_demo2_6789844810353091540_a.b .c {
+  width: 50px;
+}
+  "#;
+    assert_eq!(
+        simple_compare(target_code.to_string()),
+        simple_compare(res2)
+    );
+
+    let case_3 = path_resolve("tests/fixture/demo3.css");
+    let res3 = test_css_modules(case_3.as_str());
+    let target_code = r#"
+.x .fixture_demo3_16401825995056589139_b > .fixture_demo3_16401825995056589139_c  {
+  width: 20px;
+}
+  "#;
+    assert_eq!(
+        simple_compare(target_code.to_string()),
+        simple_compare(res3)
+    );
+}
```

Comment:
Let's avoid inline tests and use snapshots here, because it will be hard to maintance (if we improve or change logic we need manually change all hashes, it is really uncomfortable)